src/stores/widgets/CatalogWidgetStore.ts

```diff
@@ -64,22 +81,432 @@ export class CatalogWidgetStore {
+        this.sizeScalingType = FrameScaling.LINEAR;
+        this.sizeMin = {area: 100, diameter: 5};
+        this.sizeMax = {area: 200, diameter: 20};
+        this.sizeColumnMin = {default: undefined, clipd: undefined};
+        this.sizeColumnMax = {default: undefined, clipd: undefined};
+        this.sizeAxisTabId = CatalogSettingsTabs.SIZE_MAJOR;
+        this.sizeMinorMapColumn = CatalogOverlay.NONE;
+        this.sizeMinorArea = false;
+        this.sizeMinorScalingType = FrameScaling.LINEAR;
+        this.sizeMinorColumnMin = {default: undefined, clipd: undefined};
+        this.sizeMinorColumnMax = {default: undefined, clipd: undefined};
+        this.colorMapColumn = CatalogOverlay.NONE;
+        this.colorColumnMax = {default: undefined, clipd: undefined};
+        this.colorColumnMin = {default: undefined, clipd: undefined};
+        this.colorMap = "viridis";
+        this.colorScalingType = FrameScaling.LINEAR;
+        this.invertedColorMap = false;
+        this.orientationMapColumn = CatalogOverlay.NONE;
+        this.orientationMax = {default: undefined, clipd: undefined};
+        this.orientationMin = {default: undefined, clipd: undefined};
+        this.orientationScalingType = FrameScaling.LINEAR;
+        this.angleMax = 360;
+        this.angleMin = CatalogWidgetStore.MinAngle;
+        this.sizeColumnMinLocked = false;
+        this.sizeColumnMaxLocked = false;
+        this.sizeMinorColumnMinLocked = false;
+        this.sizeMinorColumnMaxLocked = false;
+
+        reaction(() => this.sizeMapData, (column) => {
+            const result = minMaxArray(column);
+            this.setSizeColumnMin(isFinite(result.minVal)? result.minVal : 0, "default");
+            this.setSizeColumnMax(isFinite(result.maxVal)? result.maxVal : 0, "default");
+        });
+
+        reaction(() => this.sizeArray(), (size) => {
+            if (size.length) {
+                CatalogWebGLService.Instance.updateDataTexture(this.catalogFileId, size, CatalogTextureType.Size);   
+            }
+        });
+
+        reaction(() => this.sizeColumnMin.clipd, (sizeColumnMin) => {
+            if (this.sizeColumnMinLocked) {
+                this.sizeMinorColumnMin.clipd = sizeColumnMin;
+            }
+        });
+
+        reaction(() => this.sizeColumnMax.clipd, (sizeColumnMax) => {
+            if (this.sizeColumnMaxLocked) {
+                this.sizeMinorColumnMax.clipd = sizeColumnMax;
+            }
+        });
+
+        reaction(() => this.sizeMinorMapData, (column) => {
+            const result = minMaxArray(column);
+            this.setSizeMinorColumnMin(isFinite(result.minVal)? result.minVal : 0, "default");
+            this.setSizeMinorColumnMax(isFinite(result.maxVal)? result.maxVal : 0, "default");
+        });
+
+        reaction(() => this.sizeMinorArray(), (size) => {
+            if (size.length) {
+                CatalogWebGLService.Instance.updateDataTexture(this.catalogFileId, size, CatalogTextureType.SizeMinor);
```

Comment:
would it not be simpler to just copy the column data itself, and then do the scaling/clamping on the GPU in the vertex shader? that would mean changing any of the parameters would be as simple as just changing a `uniform` parameter, instead of an expensive re-calculation and re-upload